src/zui/ZUICreatePerson/index.tsx

```diff
@@ -0,0 +1,200 @@
+        {customFields.length === 0 && (
+          <Box
+            sx={{ display: 'flex', justifyContent: 'center', m: 8, pr: '40px' }}
+          >
+            <CircularProgress />
+          </Box>
+        )}
+        {customFields.length > 0 && (
+          <PersonalInfoForm
+            onChange={(field, value) => {
+              if (value === '') {
+                const copied = { ...personalInfo };
+                delete copied[field];
+                setPersonalInfo(copied);
+              } else {
+                if (field === 'tags' && value) {
+                  setTags((prev) =>
+                    tags.includes(value as number)
+                      ? tags.filter((item) => item !== value)
+                      : [...prev, value as number]
+                  );
+                } else {
+                  setPersonalInfo((prev) => {
+                    return { ...prev, [field]: value as string };
+                  });
+                }
+              }
+            }}
+            personalInfo={personalInfo}
+            tags={tags}
+          />
+        )}
+        <Box sx={{ pr: '40px' }}>
+          <Divider />
+          <Box
+            alignItems="center"
+            display="flex"
+            justifyContent="flex-end"
+            mt={2}
+          >
+            <Box>
+              <Button
+                onClick={() => {
+                  onClose();
+                  setPersonalInfo({});
+                  setTags([]);
+                }}
+                sx={{ mr: 2 }}
+                variant="text"
+              >
+                <Msg id={messageIds.createPerson.cancel} />
+              </Button>
+              <Button
+                disabled={
+                  personalInfo.first_name === undefined ||
+                  personalInfo.last_name === undefined ||
+                  checkInvalidFields(customFields, personalInfo).length !== 0
+                }
+                onClick={async (e) => {
+                  const person = await createPerson(personalInfo, tags);
+                  if (onSubmit) {
+                    onSubmit(e, person);
+                  }
+                  onClose();
+                  setPersonalInfo({});
+                  setTags([]);
+                }}
+                variant="contained"
+              >
+                {submitLabel ?? (
+                  <Msg id={messageIds.createPerson.submitLabel.default} />
+                )}
+              </Button>
+            </Box>
+          </Box>
+        </Box>
+      </Box>
+    </Dialog>
+  );
+};
+
+export default ZUICreatePerson;
+
+export const checkInvalidFields = (
+  customFields: ZetkinCustomField[],
+  personalInfo: ZetkinCreatePerson
+) => {
+  let invalidFields: string[] = [];
+
+  const customFieldURLSlugs = customFields
+    .filter((item) => item.type === 'url')
+    .map((item) => item.slug);
+
+  //email
+  if (!isEmail(personalInfo.email || '') && personalInfo.email !== undefined) {
```

Comment:
Is there any reason why this couldn't be simplified like below?

```suggestion
  if (personalInfo.email && !isEmail(personalInfo.email || '')) {
```